go-controller/pkg/ovn/pods_test.go

```diff
@@ -715,6 +715,14 @@ var _ = Describe("OVN Pod Operations", func() {
 				// Update namespace to remove annotation
 				namespaceT.Annotations = nil
 				_, err = fakeOvn.fakeClient.CoreV1().Namespaces().Update(&namespaceT)
+				Expect(err).NotTo(HaveOccurred())
+				Eventually(func() map[string]string {
+					updatedNs, err := fakeOvn.fakeClient.CoreV1().Namespaces().Get(namespaceT.Name, metav1.GetOptions{})
+					if err != nil {
+						return map[string]string{"ns": "error"}
+					}
+					return updatedNs.Annotations
+				}).Should(BeEmpty())
 				// Create new pod
```

Comment:
@dcbw sorry I should have put a comment on this PR yesterday, but we actually need (cringe) a time.Sleep here for 1 second. This fixed the issue because in the background we cannot be certain when nsInfo is updated because it has to wait on a lock. So even though k8s API will show hte namespace annotations are updated there is a lag in ovn-k8s before its updated. I don't really think there's much we can do about it, other than add a little time before we create the new pod.